### PR TITLE
Manifests for promotion of CSI-proxy

### PR DESCRIPTION
**Pull request by jingxu97**
This describes the non-image artifacts that should be published as
part of csiproxy.

The automated promotion job is not yet in place, so promotion will be
done manually or will have to wait for the prow job.

### CODE REVIEW

**Comment by jingxu97** [comment]
@justinsb I tried to follow the PR for kops https://github.com/kubernetes/k8s.io/pull/1043/files to promote csi-proxy binary.

Is there a way to automate this process or still need to be manually done? Thanks!

**Comment by spiffxp** [comment]
I unfortunately don't know the manual process that is involved here, does anyone from @kubernetes/release-engineering know what's being referenced here?

**Comment by cpanato** [comment]
I'm not aware of anything to automate. also this is kind new to me, but if need I can investigate and see what we can do

**Comment by spiffxp** [comment]
@justinsb 
>then to run the robot binary "by hand"

Which binary? What is/are the commandline invocation(s) you use?

**Comment by ameukam** [comment]
> @justinsb
> 
> > then to run the robot binary "by hand"
> 
> Which binary? What is/are the commandline invocation(s) you use?

I think it's `kpromo` mentioned in https://github.com/kubernetes/k8s.io/issues/2624.

**Comment by justaugustus** [comment]
I left an update at https://github.com/kubernetes/k8s.io/issues/2624#issuecomment-909858376, but want to echo it here as well:

> Status comment is [here](https://github.com/kubernetes/k8s.io/issues/2624#issuecomment-909247637), but just to say a little more...
> 
> I've restored all of the promotion tools to the CIP repo, which is easier to do now that bazel has been removed (thanks @tylerferrara + @listx).
> 
> `kpromo` images build in presubmit (test env) and postsubmit (staging env).
> 
> There's a test periodic job for file promotion, which is analogous to the image promoter periodic (but uses debug values).
> 
> ...
> 
> tl;dr -- I see a few things to work on/ask questions about:
> 
>     * Ensuring restrictions on top-level prod writes
> 
>     * Support for multiple manifests (all staging projects) and multiple releases (per project)
> 
>     * Enable manifests with artifacts that no longer exist on staging buckets to be skipped (we expect the staging artifacts to age out)
>       
>       * We should still report the issue, and:
>       * Probably still call this a failure if any of the assets in a single manifest are not found i.e., if there were supposed to be 10 files in a 1.16 release but we only find 1, the copy should be skipped and reported on
> 
>     * Should we allow backfill operations on a directory i.e., if the staging directory initially had 10 objects, but now has 15, do we care that 5 objects will be promoted on the next run?
>       
>       * Should this be configurable per project?
> 
> 
> I'll be working on some of this in [kubernetes-sigs/k8s-container-image-promoter#409](https://github.com/kubernetes-sigs/k8s-container-image-promoter/pull/409).

**Comment by justaugustus** [comment]
/test all
(To test https://github.com/kubernetes/test-infra/pull/23491)

**Comment by jingxu97** [comment]
@justinsb  Could you please take a look at it again?

File: artifacts/filestores/k8s-staging-sig-storage/filepromoter-manifest.yaml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+# google group for gcr.io/k8s-staging-kops is k8s-infra-staging-kops@kubernetes.io
```
[the remark this answers is not in the document]
Comment (jingxu97) [reply artifacts/filestores/k8s-staging-sig-storage/filepromoter-manifest.yaml]: fixed

File: artifacts/filestores/k8s-staging-sig-storage/filepromoter-manifest.yaml [hunk 2]
```diff
@@ -0,0 +1,6 @@
+# google group for gcr.io/k8s-staging-sig-storage is kubernetes-sig-storage@kubernetes.io
+filestores:
+- base: gs://k8s-staging-sig-storage/csiproxy/
+  src: true
+- base: gs://k8s-artifacts-prod/binaries/storage/csiproxy
+  service-account: k8s-infra-gcr-promoter@k8s-artifacts-prod.iam.gserviceaccount.com
```
Comment (justaugustus) [inline artifacts/filestores/k8s-staging-sig-storage/filepromoter-manifest.yaml:6]:
```suggestion
  service-account: k8s-infra-promoter@k8s-artifacts-prod.iam.gserviceaccount.com
```

**Review by justaugustus — CHANGES_REQUESTED** [review]
@jingxu97 -- I've added you as a reviewer to the PR which adds docs for file/artifact promotion: https://github.com/kubernetes/k8s.io/pull/2711

Please leave some comments there if there are some specific sections I can further clarify.

On to the review, which echoes some of the guidance in the docs...

---

There are two groups defined for SIG Storage in the groups reconciler, which look like they could be well-suited for this:
- k8s-infra-staging-sig-storage@kubernetes.io: https://github.com/kubernetes/k8s.io/blob/1210db6925824ec445475560d0cdeaac94d0c4d3/groups/sig-storage/groups.yaml#L22-L38
- k8s-infra-staging-csi@kubernetes.io: https://github.com/kubernetes/k8s.io/blob/1210db6925824ec445475560d0cdeaac94d0c4d3/groups/sig-storage/groups.yaml#L40-L52

@kubernetes/sig-storage-feature-requests @kubernetes/csi-api-maintainers -- Which one should we use to delegate access here?

Does the source bucket (`gs://k8s-staging-sig-storage/csiproxy/`) already exist?

Depending on the answers to this, I can leave more suggestions in a follow-up review.

**Comment by jingxu97** [comment]
> @jingxu97 -- I've added you as a reviewer to the PR which adds docs for file/artifact promotion: #2711
> 
> Please leave some comments there if there are some specific sections I can further clarify.
> 
> On to the review, which echoes some of the guidance in the docs...
> 
> There are two groups defined for SIG Storage in the groups reconciler, which look like they could be well-suited for this:
> 
> * [k8s-infra-staging-sig-storage@kubernetes.io](mailto:k8s-infra-staging-sig-storage@kubernetes.io): https://github.com/kubernetes/k8s.io/blob/1210db6925824ec445475560d0cdeaac94d0c4d3/groups/sig-storage/groups.yaml#L22-L38
> * [k8s-infra-staging-csi@kubernetes.io](mailto:k8s-infra-staging-csi@kubernetes.io): https://github.com/kubernetes/k8s.io/blob/1210db6925824ec445475560d0cdeaac94d0c4d3/groups/sig-storage/groups.yaml#L40-L52
> 
> @kubernetes/sig-storage-feature-requests @kubernetes/csi-api-maintainers -- Which one should we use to delegate access here?
> 
> Does the source bucket (`gs://k8s-staging-sig-storage/csiproxy/`) already exist?
> 
> Depending on the answers to this, I can leave more suggestions in a follow-up review.

@justaugustus sorry for the delay. I think k8s-infra-staging-csi  group should be used in this case. gs://k8s-staging-sig-storage/csiproxy/ already exists.

File: artifacts/filestores/k8s-staging-sig-storage/filepromoter-manifest.yaml (hunk 2 above)
Comment (jingxu97) [reply artifacts/filestores/k8s-staging-sig-storage/filepromoter-manifest.yaml:6]: updated

**Comment by jingxu97** [comment]
@justaugustus  Could you please help review this again? Thank you!

File: artifacts/manifests/k8s-staging-sig-storage/master.yaml [hunk 3]
```diff
@@ -0,0 +1,3 @@
+files:
+- name: 1.0.0/csi-proxy.exe
```
Comment (msau42) [inline artifacts/manifests/k8s-staging-sig-storage/master.yaml:2]: We have released a 1.0.1 now right?

Comment (jingxu97) [reply artifacts/manifests/k8s-staging-sig-storage/master.yaml]: right, thanks for the review. updated the version to 1.0.2, the latest.

File: artifacts/manifests/k8s-staging-sig-storage/master.yaml [hunk 4]
```diff
@@ -0,0 +1,3 @@
+files:
+- name: 1.0.2/csi-proxy.exe
```
Comment (msau42) [inline artifacts/manifests/k8s-staging-sig-storage/master.yaml:2]: Are there different architectures for windows binaries?

Comment (jingxu97) [reply artifacts/manifests/k8s-staging-sig-storage/master.yaml:2]: I think we only support amd64 for Windows.

Comment (msau42) [reply artifacts/manifests/k8s-staging-sig-storage/master.yaml:2]: Do we want to add the architecture to the file name in case we support more architectures in the future?

Comment (jingxu97) [reply artifacts/manifests/k8s-staging-sig-storage/master.yaml:2]: there is no plan to support yet. But I think it is better to add it for future.
